app/models/hackathon/subscription/status.rb

```diff
@@ -9,6 +9,30 @@ module Hackathon::Subscription::Status
     scope :active_for, ->(user) { active.where(subscriber: user) }
   end
 
+  # Unsubscribe URLs must be valid for at least 30 days.
+  # https://www.ftc.gov/business-guidance/resources/can-spam-act-compliance-guide-business
+  UNSUBSCRIBE_EXPIRATION = 2.months
+
+  class_methods do
+    def manage_subscriptions_url_for(user)
+      user_signature = user.signed_id purpose: :manage_subscriptions, expires_in: UNSUBSCRIBE_EXPIRATION
+      Rails.application.routes.url_helpers.user_subscriptions_url(user_signature)
+    end
+
+    def unsubscribe_all_url_for(user)
+      user_signature = user.signed_id purpose: :manage_subscriptions, expires_in: UNSUBSCRIBE_EXPIRATION
+      Rails.application.routes.url_helpers.unsubscribe_all_user_subscriptions_url(user_signature)
+    end
+  end
+
+  def unsubscribe!
+    update(status: :inactive)
+  end
+
+  def resubscribe!
+    update(status: :active)
+  end
```

Comment:
```suggestion
  def unsubscribe
    update!(status: :inactive)
  end

  def resubscribe
    update!(status: :active)
  end
```
Maybe we could use events to record these actions as well?

Reply:
here we go! [`d1debf1` (#85)](https://github.com/hackclub/hackathons-backend/pull/85/commits/d1debf19351c0167b593a80f911c1c0413657202)

---

I'd prefer to keep it a bang method to make it more obvious that we're updating the model.

As for `update` vs `update!`, I find the non-error version to be better in this scenario since much of our use cases is bulk editing subscriptions; which we wouldn't want to error on.

```ruby
@subscriptions.each(&:resubscribe!)
# If there's already an active subscription causing it's unable to update status to active (due to validations), then i'd prefer for it silently fail
```

If we do want it to error, chances are, we're only working with one object and can simply do
`@subscription.update!(status: :active)`

Reply:
decided it would be best to un-bangify them!

[`60f18f6` (#85)](https://github.com/hackclub/hackathons-backend/pull/85/commits/60f18f64ea905a695ed26391d6c37af91de9c3af)

Reply:
I'm pretty sure that if there's only supposed to be a bang method if there's a safer version of it. Like `update` vs `update!`